### PR TITLE
function to return schema elements by id_prefixes 

### PR DESCRIPTION
**Pull request by sierra-moxon**
- remove extra refactoring from PR #35

### CODE REVIEW

File: linkml_runtime/utils/schemaview.py [hunk 1]
```diff
@@ -529,6 +531,40 @@ def expand_curie(self, uri: str) -> str:
                     return ns[pfx] + local_id
         return uri
 
+    @lru_cache(CACHE_SIZE)
+    def get_element_by_prefix(
```
Comment (cmungall) [inline linkml_runtime/utils/schemaview.py:535]: perhaps decompose this:

 - a method that takes an actual prefix as input and returns element names, `get_elements_applicable_for_prefix`
 - a method that splits CURIEs (or URIs?) and. extracts the prefix
 - a wrapper that composes these `get_elements_applicable_for_identifier`

Comment (sierra-moxon) [reply linkml_runtime/utils/schemaview.py]: I went ahead and broke this up into 3 methods and added support (by utilizing an existing function! - thanks for the pointer :) ) for URIs as well.

File: linkml_runtime/utils/schemaview.py [hunk 2]
```diff
@@ -663,6 +666,54 @@ def expand_curie(self, uri: str) -> str:
...
+            The prefix of a CURIE
+
+        Returns
+        -------
+        Optional[str]
+                The model element corresponding to the given prefix as available via
+                the id_prefixes mapped to that element.
+
+        """
+        categories = []
```
Comment (cmungall) [inline linkml_runtime/utils/schemaview.py:709]: minor: categories is a biolinkism - use elements instead? not a big deal as this is not exposed to the caller but still good to use internally consistent variable names

File: linkml_runtime/utils/schemaview.py [hunk 3]
```diff
@@ -663,6 +666,54 @@ def expand_curie(self, uri: str) -> str:
                     return ns[pfx] + local_id
         return uri
 
+    @lru_cache(CACHE_SIZE)
+    def get_elements_applicable_by_identifier(self, identifier: str) -> List[str]:
+        """
+        Get a Model element by prefix.
+
+        Parameters
```
Comment (cmungall) [inline linkml_runtime/utils/schemaview.py:674]: minor: can you change to the rst format used for other methods?

**Review by cmungall — APPROVED** [review]

**Comment by sierra-moxon** [comment]
fixed up doc and moved categories -> elements

**Pull request merged**